src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,38 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.LinkedList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List getAll(Class<? extends Machine> type) {
```

Comment:
```suggestion
    public List<Machine> getAll(Class<? extends Machine> type) {
```

Reply:
Can we do this? We return lists of bulldozers, excavators and trucks. List<Machine> != List<Bulldozer>, isn't it? This is invariance